core/conn/odbc/src/odbc/nsksrvr/Interface/Listener_srvr.cpp

```diff
@@ -81,4 +81,46 @@ void CNSKListenerSrvr::TCP_PROCESSNAME_PORT(FILE* fp)
 	fprintf(fp,"<==========TCP/PORT (%s/%d)==========>\n",m_TcpProcessName, m_port);
 }
 
+void CNSKListenerSrvr::TCP_SetKeepalive(int socketnum,
+                                        char *keepaliveStatus,
+                                        int idleTime,
+                                        int intervalTime,
+                                        int retryCount)
+{
+    //all need to be configured
+    if(NULL == keepaliveStatus){
+        return;
+    }
+    if(0 == strcmp(keepaliveStatus,"default")){
+        keepaliveOpt.isKeepalive = DEFAULT_KEEPALIVE;
+        keepaliveOpt.keepaliveIdle = DEFAULT_KEEPALIVE_TIMESEC;
+        keepaliveOpt.keepaliveInterval = DEFAULT_KEEPALIVE_INTVL;
+        keepaliveOpt.keepCount = DEFAULT_KEEPALIVE_COUNT;
+    }else
+    if(0 == strcmp(keepaliveStatus,"unenable")){
+                keepaliveOpt.isKeepalive = 0;
+                keepaliveOpt.keepaliveIdle = DEFAULT_KEEPALIVE_TIMESEC;
+                keepaliveOpt.keepaliveInterval = DEFAULT_KEEPALIVE_INTVL;
+                keepaliveOpt.keepCount = DEFAULT_KEEPALIVE_COUNT;
+    }else
+    if(0 == strcmp(keepaliveStatus, "enable")){
+        keepaliveOpt.isKeepalive = 1;
+        keepaliveOpt.keepaliveIdle = idleTime;
+        keepaliveOpt.keepaliveInterval = intervalTime;
+        keepaliveOpt.keepCount = retryCount;
+
+    }else{
+        keepaliveOpt.isKeepalive = 0;
```

Comment:
Why are the other members (keepaliveIdle and so on) not set in this code path? (I assume it matters since we set them in the "unenable" code path.)